library/core/src/primitive_docs.rs

```diff
@@ -268,6 +268,47 @@ mod prim_bool {}
 /// [`Debug`]: fmt::Debug
 /// [`default()`]: Default::default
 ///
+/// # Never type fallback
+///
+/// When the compiler sees a value of type `!` it implicitly inserts a coercion (if possible),
+/// to allow type check to infer any type:
```

Comment:
"allow type check to" is a somewhat odd phrasing, as "type check" is a verb? I think we should try to refer to "typechecking", as the gerund form is more clearly "a phase where things happen", or "the type checker", which is the thing driving the things happening:
```suggestion
/// to allow typechecking to infer any type:
```

Reply:
Yeah, so what I had in mind is `typeck`, which I expanded as "type check", instead of "type checker".
```suggestion
/// to allow type checker to infer any type:
```